File: sol/solution/0001-0100/0038. Count and Say/Solution.dart
class Solution {
  String countAndSay(int n) {
    String res = "1";
    for (int i = 2; i <= n; i++) {
      String cur = "";
      int cnt = 1;
      for (int j = 1; j < res.length; j++) {
        if (res[j] == res[j-1]) {
          cnt++;
        } else {
          cur += cnt.toString() + res[j-1];
          cnt = 1;
        }
      }
      cur += cnt.toString() + res[res.length - 1];
      res = cur;
    }
    return res;
  }
}

File: sol/solution/0001-0100/0038. Count and Say/Solution.kt
class Solution {
    fun countAndSay(n: Int): String {
        var res = "1"
        for (i in 2..n) {
            var cur = StringBuilder()
            var cnt = 1
            for (j in 1 until res.length) {
                if (res[j] == res[j-1]) cnt++
                else {
                    cur.append(cnt).append(res[j-1])
                    cnt = 1
                }
            }
            cur.append(cnt).append(res.last())
            res = cur.toString()
        }
        return res
    }
}